lib/ansible/modules/web_infrastructure/jenkins_plugin.py

```diff
@@ -526,13 +526,8 @@ def install(self):
             params.update(self.params)
             file_args = self.module.load_file_common_arguments(params)
 
-            if not self.module.check_mode:
-                # Not sure how to run this in the check mode
```

Comment:
`set_fs_attributes_if_different` is to check-aware. Please remove this (and the bellow) comment completely. And I would invert the condition to spare the `not` word.

Reply:
@jtyr Sorry to bother you. If my understanding is correct, I should use `set_fs_attributes_if_different` only in check mode?

I thought `set_fs_attributes_if_different` supports both check mode and run mode, so I removed the if-else block and fixed the code to execute `set_fs_attributes_if_different` in both check mode and run mode as follows.

https://github.com/ansible/ansible/pull/33824/commits/6374f8c78d0c5efb92cc9e3ae8da2b25b624613e

```diff
+            changed = self.module.set_fs_attributes_if_different(
+                file_args, changed)
```

Is this wrong?

Reply:
Ah, I have commented on the part which gets removed ;o) In that case you are actually removing working solution because there should be no action taken if it check mode. You could only improve whether file attributes change or not if in the check mode but that would have to be done for all file attribute which I think should not be done in the module but in the `set_fs_attributes_if_different` function.